packages/nextjs/src/config/webpack.ts

```diff
@@ -50,22 +50,23 @@ const defaultSentryWebpackPluginOptions = dropUndefinedKeys({
  * @returns The function to set as the nextjs config's `webpack` value
  */
 export function constructWebpackConfigFunction(
-  userNextConfig: ExportedNextConfig = {},
+  userNextConfig: NextConfigObject = {},
   userSentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions> = {},
 ): WebpackConfigFunction {
   const newWebpackFunction = (config: WebpackConfigObject, options: BuildContext): WebpackConfigObject => {
+    // clone to avoid mutability issues
+    let newConfig = { ...config };
+
     // if we're building server code, store the webpack output path as an env variable, so we know where to look for the
     // webpack-processed version of `sentry.server.config.js` when we need it
-    if (config.target === 'node') {
-      storeServerConfigFileLocation(config);
+    if (newConfig.target === 'node') {
+      storeServerConfigFileLocation(newConfig);
     }
 
-    let newConfig = config;
-
     // if user has custom webpack config (which always takes the form of a function), run it so we have actual values to
     // work with
     if ('webpack' in userNextConfig && typeof userNextConfig.webpack === 'function') {
-      newConfig = userNextConfig.webpack(config, options);
+      newConfig = userNextConfig.webpack(newConfig, options);
```

Comment:
Should we be running `storeServerConfigFileLocation` after this logic?

Reply:
I just wanna avoid doing `let newConfig` if possible

Reply:
> Should we be running storeServerConfigFileLocation after this logic?

Either way works. I was thinking of cloning as a first step to immediately get out of the way.

> I just wanna avoid doing let newConfig if possible

Given that we don't know if we're going to reassign it or not, the only way around using 'let' would be to move the `storeServerConfigFileLocation` call (which is soon to disappear anyway) and then use a ternary, which IMHO would be harder to read than the if, since in this case the condition we're checking is kinda long.

Curiosity: Is it just out of a general principle that immutability > mutability that you want to avoid `let`, or is there some other reason?

Reply:
> Is it just out of a general principle that immutability > mutability that you want to avoid

It's more specific to this instance. I made the comment because we are passing around config objects everywhere in this webpack logic, and I don't want someone to accidentally introduce a bug in future refactors or changes.